### PR TITLE
docs(DEVELOPER.md): add how to set up hybrid mode

### PR DESCRIPTION
**Pull request by Water-Melon**
Add a section for setting up hybrid mode dev environment.
Add a step for setting `GITHUB_TOKEN` to the section of `Build and Install from source`.

[KAG-2027](https://konghq.atlassian.net/browse/KAG-2027)

[KAG-2027]: https://konghq.atlassian.net/browse/KAG-2027?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: DEVELOPER.md [hunk 1]
```diff
@@ -320,6 +327,49 @@ When developing, you can use the `Makefile` for doing the following operations:
 | `test-plugins`     | Run the plugins test suite                             |
 | `test-all`         | Run all unit + integration + plugins tests at once     |
 
+### Setup Hybrid Mode Developing Environment
```
Comment (jschmid1) [inline DEVELOPER.md:330]:
```suggestion
### Setup Hybrid Mode Development Environment
```

File: DEVELOPER.md [hunk 2]
```diff
@@ -320,6 +327,49 @@ When developing, you can use the `Makefile` for doing the following operations:
...
+
+You can follow the steps given below to set hybrid mode environment.
```
Comment (jschmid1) [inline DEVELOPER.md:332]:
```suggestion
You can follow the steps given below to setup a hybrid mode environment.
```

**Review by jschmid1 — CHANGES_REQUESTED** [review]

File: DEVELOPER.md (hunk 1 above)
Comment (Water-Melon) [reply DEVELOPER.md]: Done

File: DEVELOPER.md (hunk 2 above)
Comment (Water-Melon) [reply DEVELOPER.md]: Done

**Review by jschmid1 — APPROVED** [review]

**Pull request merged**